CLI crashes for `click<8`

# To reproduce
```
pip install click==7.1.2
pytest tests/cli
```
output
```
======================================================================= short test summary info ========================================================================
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_invalid_dataset_id_errors_out - assert 'Error: My Exception' in ''
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[True-True-2-3532] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[True-True-2-542133] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[True-True-3-3532] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[True-True-3-542133] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[True-False-2-3532] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[True-False-2-542133] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[True-False-3-3532] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[True-False-3-542133] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[False-True-2-3532] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[False-True-2-542133] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[False-True-3-3532] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[False-True-3-542133] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[False-False-2-3532] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[False-False-2-542133] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[False-False-3-3532] - AttributeError: 'bytes' object has no attribute 'exists'
FAILED tests/cli/test_dataset_cmd.py::TestDatasetDownload::test_all[False-False-3-542133] - AttributeError: 'bytes' object has no attribute 'exists'
```

# Where does the bug happen
https://github.com/BlueBrain/Atlas-Download-Tools/blob/07a250fcbce6cea1ba9c07da22f33a93bbd7212c/src/atldld/cli/dataset.py#L159-L160

# Why?
In `click<8` the `click.Path` argument type can only produce `str` or `bytes` types, but not `pathlib.Path`. Compare:

`click==7.1.2`:
https://github.com/pallets/click/blob/1784558ed7c75c65764d2a434bd9cbb206ca939d/src/click/types.py#L590-L596

`click==8`:
https://github.com/pallets/click/blob/65eceb08e392e74dcc761be2090e951274ccbe36/src/click/types.py#L819-L828

# Solution
One of the two:
* require `click>=8`
* cast `output_folder` to `pathlib.Path` explicitly

--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,7 @@ python_requires = ">=3.6"
 install_requires = [
     "Pillow",
     "appdirs",
-    "click",
+    "click>=8",
     "dataclasses; python_version < '3.7'",
     "matplotlib",
     "numpy",

